las/client.py

```diff
@@ -89,303 +90,871 @@ def _create_signing_headers(self, path: str):
+        :param max_results: Maximum number of results to be returned
+        :type max_results: int
+        :param next_token: A unique token for each page, use the returned token to retrieve the next page.
+        :type next_token: str
+        :return: Documents response from REST API
+        :rtype: dict
 
-        :param batch_id: The batch id that contains the documents of interest
-        :type batch_id: str
-        :param consent_id: An identifier to mark the owner of the document handle
-        :type consent_id: str
-        :return: Documents from REST API contained in batch <batch id>
+        :raises: :py:class:`~las.InvalidCredentialsException`, :py:class:`~las.TooManyRequestsException`,\
+ :py:class:`~las.LimitExceededException`, :py:class:`requests.exception.RequestException`
+        """
+        params = {
+            'batchId': batch_id,
+            'consentId': consent_id,
+            'maxResults': max_results,
+            'nextToken': next_token,
+        }
+        return self._make_request(requests.get, '/documents', params=params)
+
+    def delete_documents(self, consent_id: Optional[Queryparam] = None) -> dict:
+        """Delete documents with the provided consent_id, calls the DELETE /documents endpoint.
+
+        >>> from las.client import BaseClient
+        >>> client = BaseClient()
+        >>> client.delete_documents('<consent id>')
+
+        :param consent_id: Ids of the consents that marks the owner of the document handle
+        :type consent_id: Queryparam
+        :return: Documents response from REST API
         :rtype: dict
 
         :raises: :py:class:`~las.InvalidCredentialsException`, :py:class:`~las.TooManyRequestsException`,\
  :py:class:`~las.LimitExceededException`, :py:class:`requests.exception.RequestException`
         """
+        return self._make_request(requests.delete, '/documents', params={'consentId': consent_id})
+
+    def get_document(self, document_id: str) -> dict:
+        """Get document from the REST API, calls the GET /documents/{documentId} endpoint.
+
+        >>> from las.client import BaseClient
+        >>> client = BaseClient()
+        >>> client.get_document(document_id='<document id>')
+
+        :param document_id: Id of the document
+        :type document_id: str
+        :return: Document response from REST API
+        :rtype: dict
 
-        return self._make_request(requests.get, '/documents', params={'batchId': batch_id, 'consentId': consent_id})
+        :raises: :py:class:`~las.InvalidCredentialsException`, :py:class:`~las.TooManyRequestsException`,\
+ :py:class:`~las.LimitExceededException`, :py:class:`requests.exception.RequestException`
+        """
+        return self._make_request(requests.get, f'/documents/{document_id}')
 
-    def create_prediction(self, document_id: str, model_name: str, max_pages: Optional[int] = None,
+    def update_document(self, document_id: str, ground_truth: Sequence[Dict[str, Union[Optional[str], bool]]]) -> dict:
```

Comment:
Does it make sense to use the Field object here?

Reply:
This is also a difference between the client and the base client which I believe is worth to conserve.